app/compare/diff_chunk.tsx

```diff
@@ -24,38 +29,80 @@ export default class DiffChunk extends React.Component<DiffChunkProps, DiffChunk
     this.setState({ expanded: true });
   }
 
-  render() {
-    const {
-      change: { value, added, removed },
-    } = this.props;
-
-    const isUnchanged = !added && !removed;
-
-    const lines = value.trimEnd().split("\n");
-
-    const nodes = lines.map((line: string, index: number) => (
-      <div key={`${index}`} className={`diff-line ${added ? "added" : ""} ${removed ? "removed" : ""}`}>
-        <div className="plus-minus-cell">
-          {added && "+"}
-          {removed && "-"}
-        </div>
-        <div className="diff-line-content">{line}</div>
-      </div>
-    ));
-
-    if (nodes.length >= 10 && isUnchanged && !this.state.expanded) {
-      return [
-        nodes[0],
-        <OutlinedButton className="diff-line collapsed" onClick={this.onExpand.bind(this)}>
-          <div className="plus-minus-cell">
+  private renderAdded(text: string) {
+    return <ins className="added">{text}</ins>;
```

Comment:
can these just be plain old div's or spans? same with del below

Reply:
This is needed for people with blindness to use this feature -- screenreaders would announce "ins" as "added" and "del" as "removed".

I think the more accessibility we can bake in, the better -- some teams might have blind engineers and lack of accessibility may be a dealbreaker

Reply:
According to [mdn docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ins), most screenreaders don't announce ins/del elements.

My biggest worry here is different browsers styling these rarely used element types differently.

More importantly I think we need to think about accessibility holistically, and not just sprinkle it in some places. There are lots of places in our app that are not accessible to those who are red/green colorblind (from Trends to the build history grid and beyond), or those using screen-readers (we use divs/spans across the board - not semantic elements).

We'll need to go through and add aria attributes, and can even add a colorblind mode in settings to support those use cases. This is something we should definitely do once we have the team and the usage to support it, but for the time being having a product that isn't accessible and cluttered seems like the worst of both worlds.

Reply:
Thanks for the explanation -- changed to regular divs & spans, and I'll keep that philosophy on accessibility in mind when working on future components.

![image](https://user-images.githubusercontent.com/2414826/98426726-048d7880-2068-11eb-9628-681fe2301b76.png)